src/main/webapp/scripts/itinerary-overview.js

```diff
@@ -227,6 +225,50 @@ function displayRoute(start, waypoints, end) {
     );
 }
 
+/**
+ * Method that changes the travel mode that the user will be using on their trip.
+ * @param travelMode{String} one of the accepted modes of travel by displayRoute() (DRIVING,WALKING,BICYCLING,TRANSIT)
+ */
+function changeTravelMode(travelMode){
+    currentMode = travelMode;
+    displayRoute(start, waypoints, end);
+}
+
+function changeTravelModeDriving(){
+    changeTravelMode("DRIVING");
+}
+
+function changeTravelModeTransit(){
+    changeTravelMode("TRANSIT");
+}
+
+function changeTravelModeWalking(){
+    changeTravelMode("WALKING");
+}
+
+function changeTravelModeCycling(){
+    changeTravelMode("BICYCLING");
+}
```

Comment:
Why don't you just insert the appropriate travel mode string into the function call in the HTML so you only need changeTravelMode and not a function for each travel mode? E.g. `onclick=changeTravelMode("WALKING")`

Reply:
Also, since there seems to be a variable (currentMode) you assign one of four possible states (DRIVING, TRANSIT, WALKING, and BICYCLING), you may want to consider implementing an enum for better code quality as explained [here](https://www.sohamkamani.com/blog/2017/08/21/enums-in-javascript/). However, your current solution works fine so this is only a suggestion if you feel this will make your code more verbose.

Reply:
Didn't know how to do it at first so I just got a working solution because it doesn't let me put "" inside the function. Should be better now though